component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -525,6 +523,65 @@ private String getParameterFromURIString(String uriString, String parameterName)
         }
         return StringUtils.EMPTY;
     }
-}
 
+    /**
+     * Resolves the callback URL from the authenticator properties. If a callback URL is provided, that value is
+     * returned. If empty, returns the global commonauth endpoint when organization-specific commonauth is disabled,
+     * or otherwise the organization-specific commonauth endpoint.
+     *
+     * @param authenticatorProperties the authenticator properties map.
+     * @return the resolved callback URL.
+     * @throws RuntimeException if building the callback URL fails.
+     */
+    @Override
+    protected String getCallbackUrl(Map<String, String> authenticatorProperties) {
+
+        String callbackUrl = authenticatorProperties.get(Office365AuthenticatorConstants.CALLBACK_URL);
+        if (StringUtils.isNotEmpty(callbackUrl)) {
```

Comment:
Previously if we pass " " as the callbackUrl it will return the IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, true, true) but now it will return " ". Hence we should change to StringUtils.isNotBlank(callbackUrl) instead of StringUtils.isNotEmpty(callbackUrl)

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/38/commits/245f529e92153837353ab9118edca162af3785eb.